teradata/datadog_checks/teradata/check.py

```diff
@@ -0,0 +1,168 @@
+            'teradata_server:{}'.format(self.instance.get('server')),
+            'teradata_port:{}'.format(self.instance.get('port', 1025)),
+        ]
+        self._tags.extend(self.instance.get('tags', []))
+
+        manager_queries = []
+        if is_affirmative(self.instance.get('collect_res_usage', False)):
+            manager_queries.extend(COLLECT_RES_USAGE)
+        else:
+            manager_queries.extend(DEFAULT_QUERIES)
+
+        self._query_manager = QueryManager(
+            self,
+            self._execute_query_raw,
+            queries=manager_queries,
+            tags=self._tags,
+            error_handler=self._executor_error_handler,
+        )
+        self.check_initializations.append(self.initialize_config)
+        self.check_initializations.append(self._query_manager.compile_queries)
+
+        self._query_errors = 0
+
+    def check(self, _):
+        with self.connect() as conn:
+            if conn:
+                self._connection = conn
+                self._query_manager.execute()
+
+        self.submit_health_checks()
+
+    def initialize_config(self):
+        self._connect_params = {
+            'host': self.config.server,
+            'account': self.config.account if self.config.account else None,
+            'database': self.config.database,
+            'dbs_port': str(self.config.port),
+            'logmech': self.config.auth_mechanism.upper() if self.config.auth_mechanism else None,
+            'logdata': self.config.auth_data if self.config.auth_data else None,
+            'user': self.config.username if self.config.username else None,
+            'password': self.config.password if self.config.password else None,
+            'https_port': str(self.config.https_port) if self.config.https_port else '443',
+            'sslmode': self.config.ssl_mode.upper() if self.config.ssl_mode else 'PREFER',
+            'sslprotocol': self.config.ssl_protocol if self.config.ssl_protocol else 'TLSv1.2',
+        }
+
+    def _execute_query_raw(self, query):
+        with closing(self._connection.cursor()) as cursor:
+            query = query.format(self.config.database)
+            cursor.execute(query)
+            if cursor.rowcount < 1:
+                self._query_errors += 1
+                self.log.warning('Failed to fetch records from query: `%s`.', query)
+                return None
+            for row in cursor.fetchall():
+                try:
+                    yield self._timestamp_validator(row, query)
+                except Exception:
+                    self.log.debug('Unable to validate Resource Usage View timestamp, skipping row.')
+                    yield row
+
+    def _executor_error_handler(self, error):
+        self._query_errors += 1
+        if self._connection:
+            try:
+                self._connection.close()
+            except Exception as e:
+                self.log.warning("Couldn't close the connection after a query failure: %s", str(e))
```

Comment:
I think this should be debug. It's not very useful information for customers

Reply:
That makes sense! I took this part out actually, because it was closing the connection before the executor could finish running the remaining queries.